packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -128,10 +128,21 @@ export default createRule({
 
           caseTypes.add(getNodeType(switchCase.test));
         }
+        const missingBranchTypes = unionTypes.filter(unionType => {
+          const type = isTypeFlagSet(
+            unionType,
+            ts.TypeFlags.Number |
+              ts.TypeFlags.String |
+              ts.TypeFlags.Object |
+              ts.TypeFlags.Intersection,
+          );
+
+          if (type) {
+            return caseTypes.has(unionType);
+          }
 
-        const missingBranchTypes = unionTypes.filter(
-          unionType => !caseTypes.has(unionType),
-        );
+          return !caseTypes.has(unionType);
```

Comment:
Both `type` and `caseTypes.has` return `boolean. Just use `type === caseTypes.has(unionType)` here (poor man's XNOR)